src/Shared/CertificateGeneration/CertificateManager.cs

```diff
@@ -44,9 +44,13 @@ internal class CertificateManager
 
         // Setting to 0 means we don't append the version byte,
         // which is what all machines currently have.
-        public int AspNetHttpsCertificateVersion { get; set; } = 1;
+        public static int AspNetHttpsCertificateVersion { get; set; } = 1;
```

Comment:
Ugh, public static writable properties?

Reply:
Yeah, it's a bit ugly, I didn't want to make a big change out of this. 

It's not public code as the class is internal and it is "Shared source". It has become a bit of a mess, over time, but it hasn't reach that point/level where its painful enough to deserve cleanup.

But certainly we can look at cleaning this up as part of 5.0 if we decide to tackle additional things with this tool (like better tracing or adding telemetry to it, which I think would be very useful in the long run (none of our tools currently has))